crates/turborepo-lib/src/shim.rs

```diff
@@ -792,306 +724,149 @@ mod test {
+        //   single_root/
+        //     package.json
+        //     src/
+        let (_tmp, tmp_dir) = tmp_dir();
+        let irrelevant = tmp_dir.join_component("irrelevant");
+        irrelevant.create_dir_all().unwrap();
+        let monorepo_root = tmp_dir.join_component("monorepo_root");
+        let monorepo_pkg_json = monorepo_root.join_component("package.json");
+        monorepo_pkg_json.ensure_dir().unwrap();
+        monorepo_pkg_json
+            .create_with_contents("{\"workspaces\": [\"packages/*\"]}")
+            .unwrap();
+        monorepo_root
+            .join_component("package-lock.json")
+            .create_with_contents("")
+            .unwrap();
 
-        let (_tmp, root) = tmp_dir();
-        let root_one = root.join_components(&["..", "root-one"]);
-        let root_two = root_one.join_component("root-two");
-        let project_one = root.join_components(&["..", "project-one"]);
-        let project_two = project_one.join_component("project-two");
+        let app_1 = monorepo_root.join_components(&["packages", "app-1"]);
+        let app_1_pkg_json = app_1.join_component("package.json");
+        app_1_pkg_json.ensure_dir().unwrap();
+        app_1_pkg_json
+            .create_with_contents("{\"name\": \"app_1\"}")
+            .unwrap();
+        let app_1_src = app_1.join_component("src");
+        app_1_src.create_dir_all().unwrap();
+
+        let standalone = monorepo_root.join_component("standalone");
+        let standalone_pkg_json = standalone.join_component("package.json");
+        standalone_pkg_json.ensure_dir().unwrap();
+        standalone_pkg_json
+            .create_with_contents("{\"name\":\"standalone\"}")
+            .unwrap();
+        standalone
+            .join_component("package-lock.json")
+            .create_with_contents("")
+            .unwrap();
+
+        let standalone_monorepo = monorepo_root.join_component("standalone_monorepo");
+        let app_2 = standalone_monorepo.join_components(&["packages", "app-2"]);
+        app_2.create_dir_all().unwrap();
+        app_2
+            .join_component("package.json")
+            .create_with_contents("{\"name\":\"app-2\"}")
+            .unwrap();
+        standalone_monorepo
+            .join_component("package.json")
+            .create_with_contents("{\"workspaces\": [\"packages/*\"]}")
+            .unwrap();
+        standalone_monorepo
+            .join_component("package-lock.json")
+            .create_with_contents("")
+            .unwrap();
+
+        let single_root = tmp_dir.join_component("single_root");
+        let single_root_src = single_root.join_component("src");
+        single_root_src.create_dir_all().unwrap();
+        single_root
+            .join_component("package.json")
+            .create_with_contents("{\"name\": \"single-root\"}")
+            .unwrap();
+        single_root
+            .join_component("package-lock.json")
+            .create_with_contents("")
+            .unwrap();
```

Comment:
It's nice to create fixtures at runtime instead of maintaining them in git, but this is also much harder to read (and can fall out of sync with the explanatory comment). Kind of wishing there was a DSL or some declarative instantiation

Reply:
There might be something good for this in the Rust ecosystem. I haven't really looked.